event_session/models/event_session.py

```diff
@@ -0,0 +1,524 @@
+
+    def action_open_registrations(self):
+        """Open session registrations"""
+        self.ensure_one()
+        action = self.env.ref("event.act_event_registration_from_event").read()[0]
+        action["domain"] = [("id", "in", self.registration_ids.ids)]
+        action["context"] = {
+            "default_event_id": self.event_id.id,
+            "default_session_id": self.id,
+        }
+        return action
+
+    def action_set_done(self):
+        """Similar to core's :meth:`event_event.action_set_done`"""
+        first_ended_stage = self.env["event.stage"].search(
+            [("pipe_end", "=", True)], limit=1, order="sequence"
+        )
+        if first_ended_stage:
+            self.stage_id = first_ended_stage
+
+    def _get_ics_file(self):
+        """Similar to core's :meth:`event_event._get_ics_file`"""
+        result = {}
+        if not vobject:  # pragma: no cover
+            return result
+        for rec in self:
+            cal = vobject.iCalendar()
+            cal_event = cal.add("vevent")
+            cal_event.add("created").value = fields.Datetime.now().replace(
+                tzinfo=pytz.timezone("UTC")
+            )
+            cal_event.add("dtstart").value = fields.Datetime.from_string(
+                rec.date_begin
+            ).replace(tzinfo=pytz.timezone("UTC"))
+            cal_event.add("dtend").value = fields.Datetime.from_string(
+                rec.date_end
+            ).replace(tzinfo=pytz.timezone("UTC"))
+            cal_event.add("summary").value = rec.name
+            if rec.address_id:
+                cal_event.add("location").value = rec.sudo().address_id.contact_address
+            result[rec.id] = cal.serialize().encode("utf-8")
+        return result
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+        # Subscribe the organizer to sessions. Similar to core's behaviour for events.
+        for rec in records:
+            if rec.organizer_id:
+                rec.message_subscribe([rec.organizer_id.id])
+        return records
+
+    @api.model
+    def _session_update_fields(self):
+        """List of fields that could be synced with session_update"""
+        return ["active"]
+
+    def _compute_session_update_message(self):
+        """Human readable list of fields that could be synced with session_update"""
+        fnames = self._session_update_fields()
+        fdescs = map(lambda fname: self._fields[fname].string, fnames)
+        self.session_update_message = "\n".join(map(lambda s: f"* {s}", fdescs))
+
+    def _sync_session_update(self, vals):
+        """Handles write on multiple sessions at once from the UX"""
+        update = vals.pop("session_update", "this")
+        if update not in ("subsequent", "all"):
+            return
+        if len(self) > 1:
+            raise ValidationError(
+                _("You cannot use session_update when writing on recordsets")
+            )
```

Comment:
I'm just wondering why we are not using `self.ensure_one()` here ?

```suggestion
        self.ensure_one()
```

Reply:
Well, ensure_one will fail if len == 0, which may happen sometimes due to programatically setting field values on empty recordset (that shouldn't fail, just shouldn't do anything)

But more importantly, I like to raise a clear error message